check_api/src/main/java/com/google/errorprone/util/SourceVersion.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2023 The Error Prone Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.errorprone.util;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
+import com.sun.tools.javac.code.Source;
+import com.sun.tools.javac.code.Source.Feature;
+import com.sun.tools.javac.util.Context;
+import java.util.Arrays;
+
+/**
+ * JDK source version utilities.
+ *
+ * @see RuntimeVersion
+ */
+public final class SourceVersion {
+  private static final ImmutableMap<String, Feature> KNOWN_FEATURES =
+      Maps.uniqueIndex(Arrays.asList(Feature.values()), Enum::name);
+
+  /** Returns true if the compiler source version level supports switch expressions. */
+  public static boolean supportsSwitchExpressions(Context context) {
+    return supportsFeature("SWITCH_EXPRESSION", context);
+  }
+
+  /** Returns true if the compiler source version level supports text blocks. */
+  public static boolean supportsTextBlocks(Context context) {
+    return supportsFeature("TEXT_BLOCKS", context);
+  }
```

Comment:
The reason for adding this method is twofold:
1. With two examples it should be clear for future devs how other methods may be added.
2. I have a use case for this particular feature in PicnicSupermarket/error-prone-support#198.

Reply:
👍 

There is at least one other use of `Source` in Error Prone that can also be cleaned up to use this approach. (That's just an FYI, it doesn't need to happen in this PR.)

https://cs.github.com/google/error-prone/blob/721096fd16b9f9af7ae7dd657bd9d4ae75db5b46/core/src/main/java/com/google/errorprone/bugpatterns/VarChecker.java?q=repo%3Agoogle%2Ferror-prone++Source.instance#L102